lib/src/rollout_evaluator.dart

```diff
@@ -1,349 +1,1071 @@
+    }
+
+    evaluateLogger?.logPrerequisiteFlagEvaluationStart(prerequisiteFlagKey);
+
+    EvaluationContext prerequisiteFlagContext = EvaluationContext(
+        prerequisiteFlagKey,
+        evaluationContext.user,
+        visitedKeys,
+        evaluationContext.settings);
+
+    EvaluationResult evaluateResult = _evaluateSetting(
+        prerequisiteFlagSetting, prerequisiteFlagContext, evaluateLogger);
+
+    visitedKeys.remove(evaluationContext.key);
+
+    PrerequisiteComparator prerequisiteComparator =
+        PrerequisiteComparator.values.firstWhere(
+            (element) =>
+                element.id == prerequisiteFlagCondition.prerequisiteComparator,
+            orElse: () => throw ArgumentError(
+                "Prerequisite Flag comparison operator is invalid."));
+    SettingsValue? conditionValue = prerequisiteFlagCondition.value;
+    bool result;
+
+    switch (prerequisiteComparator) {
+      case PrerequisiteComparator.equals:
+        result = conditionValue == evaluateResult.value;
+        break;
+      case PrerequisiteComparator.notEquals:
+        result = conditionValue != evaluateResult.value;
+        break;
+      default:
+        throw ArgumentError(
+            "Prerequisite Flag comparison operator is invalid.");
+    }
+
+    evaluateLogger?.logPrerequisiteFlagEvaluationResult(
+        prerequisiteFlagCondition, evaluateResult.value, result);
+
+    return result;
   }
 
-  String _formatValidationErrorRule(
-      {required String comparisonAttribute,
-      required String userValue,
-      required int comparator,
-      required String comparisonValue,
-      required dynamic error}) {
-    return 'Evaluating rule: [$comparisonAttribute:$userValue] [${RolloutEvaluator._comparatorTexts[comparator]}] [$comparisonValue] => Skip rule. Validation error: $error';
+  EvaluationResult? _evaluatePercentageOptions(
+      List<PercentageOption> percentageOptions,
+      String? percentageOptionAttribute,
+      EvaluationContext evaluationContext,
+      TargetingRule? parentTargetingRule,
+      EvaluateLogger? evaluateLogger) {
+    ConfigCatUser? contextUser = evaluationContext.user;
+    if (contextUser == null) {
+      evaluateLogger?.logPercentageOptionUserMissing();
+      if (!evaluationContext.isUserMissing) {
+        evaluationContext.isUserMissing = true;
+        _logger.warning(3001,
+            "Cannot evaluate targeting rules and % options for setting '${evaluationContext.key}' (User Object is missing). You should pass a User Object to the evaluation methods like `getValue()`/`getValueAsync()` in order to make targeting work properly. Read more: https://configcat.com/docs/advanced/user-object/");
+      }
+      return null;
+    }
+    String percentageOptionAttributeValue;
+    String? percentageOptionAttributeName = percentageOptionAttribute;
+    if (percentageOptionAttributeName == null ||
+        percentageOptionAttributeName.isEmpty) {
```

Comment:
Let's not fall back to Identifier in the case of empty string. Reference implementations don't do this. E.g. https://github.com/configcat/python-sdk/blob/5c6f62390a00421d8b42b49de03ed66b687fb66c/configcatclient/rolloutevaluator.py#L244-L248

(Unfortunately, this issue is also present in the Java and Android SDK. Please correct those as well.)

```suggestion
    if (percentageOptionAttributeName == null) {
```